doc/conduit.rst

```diff
@@ -1,14 +1,239 @@
+
+- **Inputs**: An input is what defines *when* a step function should get scheduled for execution.
+- **Outputs**: The output is publishing the result of the execution (the *what*).
+- **Stepper**: Steppers define *how* the inputs get consumed and the outputs produced.
+- **Graph Description**: The *why* is handled by describing the connections and hierarchy for our platform.
+
+
+Putting these concepts together will result in a **Conduit Graph** describing the hierarchical structure of **Nodes**. This graph can then be used to derive a directed execution graph which is getting executed based on the defined policies and connections.
+
+Following is the definition of these core concepts and their API. Implementations of these concepts can be found in :ref:`heph.conduit.usage`.
+
+.. note:: While the primary focus of conduit is to provide an abstraction for building a computational graph, Inputs and Outputs can be used outside of graphs as well as building blocks to interface with external sources.
+
+.. _heph.conduit.concepts.inputs:
+
+Inputs
+======
+
+As an Input to a step function is a very vague statement, the most basic requirement is the ability to signal a result. As such, we define the following interface:
+
+.. literalinclude:: ../modules/conduit/include/hephaestus/conduit/basic_input.h
+   :caption: :cpp:class:`heph::conduit::BasicInput`
+   :language: cpp
+   :start-after: // APIDOC: begin
+   :end-before: // APIDOC: end
+
+This definition allows to define arbitrary implementations on how/when something can be marked as ready. Implementations of this interface can specify additional APIs on top, and provide a particular implementation for the trigger. By using the generic concept of a :ref:`heph.concurrency.concepts.sender`, we allow for maximum flexibility. This allows to implement `Inputs` for various use cases, including the retrieval of sensor data towards providing interoperability with Zenoh.
+
+.. _heph.conduit.concepts.inputs.basic:
+
+Basic Inputs
+------------
+
+Conduit provides an implementation for the following Basic Inputs:
+
+- :ref:`Periodic <heph.conduit.usage.inputs.periodic>`
+- :ref:`Conditional <heph.conduit.usage.inputs.conditional>`
+- :ref:`Generator <heph.conduit.usage.inputs.generator>`
+.. - :ref:`ZenohSubscriber <heph.conduit.usage.inputs.zenoh_subscriber>`
+.. - :ref:`ActionService <heph.conduit.usage.inputs.action_service>`
+
+All of them form a root of the execution graph and their main purpose is to provide abstractions to interact with the external components. They are not connectable to any outputs. Their produced value (if any), can only be consumed directly from the APIs the specific implementation provides. All of them can be used outside conduit nodes without further ado and provide re-usable components for other tasks.
```

Comment:
"All of them can be used outside conduit nodes without further ado and provide re-usable components for other tasks." doesnt really describe the basic input. that propertie goes for almost every code anywhere. I would try to avoid verbosity and remove this sentence

```suggestion
All of them form a root of the execution graph and their main purpose is to provide abstractions to interact with the external components. They are not connectable to any outputs. Their produced value (if any), can only be consumed directly from the APIs the specific implementation provides.
```